src/Perspex.Controls/LayoutTransformControl.cs

```diff
@@ -0,0 +1,407 @@
+            new FuncControlTemplate<LayoutTransformControl>(control =>
+        {
+            return new Decorator()
+            {
+                Child = new ContentPresenter()
+                {
+                    [~ContentPresenter.ContentProperty] = control[~ContentControl.ContentProperty]
+                }
+            };
+        });
+
+        /// <summary>
+        /// RenderTransform/MatrixTransform applied to TransformRoot.
+        /// </summary>
+        private MatrixTransform _matrixTransform;
+
+        /// <summary>
+        /// Transformation matrix corresponding to _matrixTransform.
+        /// </summary>
+        private Matrix _transformation;
+
+        /// <summary>
+        /// Actual DesiredSize of Child element (the value it returned from its MeasureOverride method).
+        /// </summary>
+        private Size _childActualSize = Size.Empty;
+
+        private Control _transformRoot;
+
+        public Control TransformRoot => _transformRoot ??
+                    (_transformRoot = this.GetVisualChildren().OfType<Control>().FirstOrDefault());
+
+        private IDisposable _transformChangedEvent = null;
+
+        private void OnLayoutTransformChanged(PerspexPropertyChangedEventArgs e)
+        {
+            var newTransform = e.NewValue as Transform;
+
+            if (_transformChangedEvent != null)
+            {
+                _transformChangedEvent.Dispose();
+                _transformChangedEvent = null;
+            }
+
+            if (newTransform != null)
+            {
+                _transformChangedEvent = Observable.FromEventPattern<EventHandler, EventArgs>(
+                                        v => newTransform.Changed += v, v => newTransform.Changed -= v)
+                                        .Subscribe(onNext: v => ApplyLayoutTransform());
+            }
+
+            ApplyLayoutTransform();
+        }
+
+        /// <summary>
+        /// Builds the visual tree for the LayoutTransformerControl when a new
+        /// template is applied.
+        /// </summary>
+        protected override void OnTemplateApplied(TemplateAppliedEventArgs e)
+        {
+            base.OnTemplateApplied(e);
+
+            _matrixTransform = new MatrixTransform();
+
+            if (null != TransformRoot)
+            {
+                TransformRoot.RenderTransform = _matrixTransform;
+                TransformRoot.TransformOrigin = new RelativePoint(0, 0, RelativeUnit.Absolute);
+            }
+
+            ApplyLayoutTransform();
+        }
+
+        /// <summary>
+        /// Applies the layout transform on the LayoutTransformerControl content.
+        /// </summary>
+        /// <remarks>
+        /// Only used in advanced scenarios (like animating the LayoutTransform).
+        /// Should be used to notify the LayoutTransformer control that some aspect
+        /// of its Transform property has changed.
```

Comment:
When animating the layout transform, `OnLayoutTransformChanged` will be called, which will call this method. Does it really need to be public?